EpikCord/utils/utils.py

```diff
@@ -137,32 +128,38 @@ def get_mime_type_for_image(data: bytes) -> str:
 
         raise InvalidArgumentType("Unsupported image type given")
 
-    def _bytes_to_base64_data(self, data: bytes) -> str:
+    @staticmethod
+    def _bytes_to_base64_data(data: bytes) -> str:
         fmt = "data:{mime};base64,{data}"
-        mime = self.get_mime_type_for_image(data)
+        mime = Utils.get_mime_type_for_image(data)
         b64 = b64encode(data).decode("ascii")
         return fmt.format(mime=mime, data=b64)
 
-    def component_from_type(self, component_data: dict):
+    @staticmethod
+    def component_from_type(component_data: dict):
+        component_types = {2: Button, 3: SelectMenu, 4: TextInput}
```

Comment:
Dictionary are a pretty data structure that could be avoided here by simply using a tuple and indexing it.

```py
component_types = (Button, SelectMenu, TextInput)

...
c_type = component_data["type"]

if 1 < c_type < 5: 
    raise ...

component_cls = component_types[c_type - 2]
```

Reply:
Good point